# **Hypothesis Testing Excercise**

## **Question 1**

**A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from bot units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.**

We are going to conduct 2 tailed t-test on 2 independent samples with numerical data



*   Null Hypothesis:- Unit A = Unit B
*   Alternate Hypothesis: - Unit A != Unit B

significance 5%

Alpha = 0.05

as it is two-tailed test



*   alpha/2 = 0.025



In [1]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

In [29]:
from google.colab import files
uploaded = files.upload()

Saving Cutlets.csv to Cutlets (1).csv


In [30]:
data = pd.read_csv('Cutlets.csv')
data.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [31]:
data.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [32]:
unit_A = pd.Series(data.iloc[:,0])
unit_A

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [33]:
unit_B = pd.Series(data.iloc[:,1])
unit_B

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [34]:
# 2-sample 2-tail ttest:   stats.ttest_ind(array1,array2)
# ind -> independent samples

p_value = stats.ttest_ind(unit_A, unit_B)
p_value

TtestResult(statistic=0.7228688704678063, pvalue=0.4722394724599501, df=68.0)

Compare p_value with Significance level
If p_value is not equal to significance level then we failed to reject Null Hypothesis because of lack of evidence.

If p_value is equal to significance then we reject Null Hypothesis

as p_value is greater than 0.5 i.e 0.025
Hence we fail to reject Null Hypothesis and there is no significant difference between the two samples.

### **Question 02**

**A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.**

**Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.**

Minitab File: LabTAT.mtw

here,

Null Hypothesis: - L1 = L2 = L3 = L4

Alternate Hypothesis:- one of them is different

here we have to perform ANOVA test

Significance 5%

alpha: - 0.05

In [17]:
from google.colab import files
uploaded = files.upload()

Saving LabTAT.csv to LabTAT.csv


In [18]:
labtat = pd.read_csv('LabTAT.csv')
labtat.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [19]:
labtat.describe()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
count    120.000000    120.000000    120.000000     120.00000
mean     178.361583    178.902917    199.913250     163.68275
std       13.173594     14.957114     16.539033      15.08508
min      138.300000    140.550000    159.690000     124.06000
25%      170.335000    168.025000    188.232500     154.05000
50%      178.530000    178.870000    199.805000     164.42500
75%      186.535000    189.112500    211.332500     172.88250
max      216.390000    217.860000    238.700000     205.18000

In [21]:
stats.f_oneway(labtat.iloc[:,0],labtat.iloc[:,1],labtat.iloc[:,2],labtat.iloc[:,3])
print('p_value =',p_value)

p_value = 2.1156708949992414e-57


So, Comparing p_value with the Significance level

We can say that there is difference in average TAT among the different laboratories.

### **Question 03**

We are going to conduct a test of Independence using Chi-Square test with contingency table


*   Null Hypothesis: There is no association or dependency between the gender based buyer rations across regions
*   Alternative Hypthosis: There is a significant association or dependency between the gender based buyer rations across regions

Significance is 5%

alpha = 0.05

In [37]:
from google.colab import files
uploaded = files.upload()

Saving BuyerRatio.csv to BuyerRatio (1).csv


In [38]:
buyer = pd.read_csv('BuyerRatio.csv')
buyer

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [39]:
buyer_table = buyer.iloc[:,1:6]
buyer_table

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [40]:
buyer_table.values

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [45]:
val = stats.chi2_contingency(buyer_table)
val

Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [46]:
type(val)

scipy.stats.contingency.Chi2ContingencyResult

In [47]:
observed = np.array([50, 142, 131, 70, 435, 1523, 1356, 750])
expected = np.array([42.76531299,  146.81287862,  131.11756787, 72.30424052, 442.23468701, 1518.18712138, 1355.88243213, 747.69575948])

In [48]:
# Comparing Evidence with Hypothesis

statistics, p_value = stats.chisquare(observed, expected, ddof = 3)
print("Statistics = ",statistics,"\n",'P_Value = ', p_value)


Statistics =  1.5959455390914483 
 P_Value =  0.8095206646905712


So, Comparing p_value with significance level
we fail to reject Null Hypothesis because of lack evidence.
Therefore, there is no association or dependency between male-female buyers rations and are similar across regions.

## **Question 04**

**TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences**

Minitab File: CustomerOrderForm.mtw

In [49]:
from google.colab import files
uploaded = files.upload()

Saving Costomer+OrderForm.csv to Costomer+OrderForm.csv


In [50]:
cust = pd.read_csv('Costomer+OrderForm.csv')
cust

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [51]:
cust.describe()

Phillippines   Indonesia       Malta       India
count           300         300         300         300
unique            2           2           2           2
top      Error Free  Error Free  Error Free  Error Free
freq            271         267         269         280

In [53]:
cust.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [54]:
cust.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [55]:
cust.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [56]:
cust.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [57]:
# Make a contingency table
obs=np.array([[271,267,269,280],[29,33,31,20]])
obs

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [59]:
# Chi2 contengency independence test

from scipy.stats import chi2_contingency
chi2_contingency(obs) # o/p is (Chi2 stats value, p_value, df, expected obsvations)


Chi2ContingencyResult(statistic=3.858960685820355, pvalue=0.2771020991233135, dof=3, expected_freq=array([[271.75, 271.75, 271.75, 271.75],
       [ 28.25,  28.25,  28.25,  28.25]]))

Significance = 0.05, p_value = 0.277

**Independent**

we fail to reject Null Hypothesis because of lack of Evidence.